app/models/faculty_survey.rb

```diff
@@ -0,0 +1,23 @@
+# == Schema Information
+#
+# Table name: faculty_surveys
+#
+#  id         :bigint           not null, primary key
+#  uniqname   :string
+#  term_id    :bigint
+#  unit_id    :bigint
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  program_id :integer
+#
+
+class FacultySurvey < ApplicationRecord
+  has_many :config_questions
+  belongs_to :term
+  belongs_to :unit
+
```

Comment:
Do we need to add association of **_has_many :faculty_surveys_** in the **_Term_** and **_Unit_** model  ?

Reply:
Yes, I will add that and I will add Terms and Units filters to the Faculty Survey index page.

Reply:
@anantaUM  I decided not to add _has_many :faculty_surveys_ to Terms and Units because I don't run queries like 

`term.faculty_surveys or unit.faculty_surveys`

I added term scopes to the FacultySurvey model to get faculty_surveys for a term.
And to get faculty_surveys for a unit we have to check if a unit is in the list of the current_user units and then if the unit has the faculty_survey preference turned on.
So each time we have to run a query like 

`FacultySurvey.where(unit_id: <something>) or  FacultySurvey.where(term_id: <something>) `

Reply:
from: https://stackoverflow.com/questions/57792841/is-it-always-best-practice-to-pair-a-belongs-to-association-with-a-matching-has

It is considered a best practice, but only because of Active Record magic. From the [Rails guide on Bi-Directional Associations](https://guides.rubyonrails.org/association_basics.html#bi-directional-associations):

"Active Record will attempt to automatically identify that these two models share a bi-directional association based on the association name. In this way, Active Record will only load one copy of the Author object, making your application more efficient and preventing inconsistent data:"

Reply:
I added has_many :faculty_surveys to Turn and Unit. Thanks for the reference.